advisor/src/main/kotlin/advisors/VulnerableCode.kt

```diff
@@ -112,11 +115,19 @@ class VulnerableCode(name: String, config: VulnerableCodeConfiguration) : Advice
      * list with a single [VulnerabilityReference] with limited data.
      */
     private fun VulnerableCodeService.VulnerabilityReference.toModel(): List<VulnerabilityReference> {
-        val sourceUri = URI(url)
+        val sourceUri = safeUri()
         if (scores.isEmpty()) return listOf(VulnerabilityReference(sourceUri, null, null))
         return scores.map { VulnerabilityReference(sourceUri, it.scoringSystem, it.value) }
     }
 
+    /**
+     * Return the URL for this reference as a URI dealing gracefully with invalid URIs. It occurred that some of the
+     * URLs returned by VulnerableCode were rejected by Java's URI parser. To prevent that in such cases the whole
+     * advisor fails, handle those URLs by returning them with a special scheme and URL-encoded.
+     */
+    private fun VulnerableCodeService.VulnerabilityReference.safeUri(): URI =
+        url.toUri().getOrDefault(URI("unparsed:${URLEncoder.encode(url, StandardCharsets.UTF_8)}"))
```

Comment:
I wonder whether having such an "unparsed" URI is of any value, or whether we rather should create a warning issue / log for such things. I currently have a tendency for the latter. What do you think @oheger-bosch?

Reply:
The problem I see here is that the vulnerability reference could also be associated with scores and severities. Consider the case that there is a reference indicating a high severity, but with an invalid URL. Wouldn't it be better than if the reference is still visible in the reports where it is expected? But since the URL is not nullable, it must contain something. The hope is that a human reader can at least get an impression from where this reference comes from.

Reply:
> Wouldn't it be better than if the reference is still visible in the reports where it is expected?

To be honest, I don't necessarily think so. First of all, personally I do not care much about the references at all; all I want to know is if there *is* a vulnerability, and I do regard all sources that ORT's advisors use as trustful anyway.

> The hope is that a human reader can at least get an impression from where this reference comes from.

With the escaping happening, a human reading is going to have a hard time making sense of the URI anyway. So I'd still prefer the issue-approach.

Reply:
Another idea could be to fixup the URI before parsing so it becomes valid. Is there a pattern behind what makes these URIs invalid?

Reply:
The solution with an issue is also fine with me, I can try this out.
I am not sure whether we can identify a general pattern. In theory, the URI conversion could always fail.

Reply:
Changed the implementation to create an issue.